src/dmd/traits.d

```diff
@@ -923,7 +923,34 @@ extern (C++) Expression semanticTraits(TraitsExp e, Scope* sc)
                     ex = dve.e1;
             }
 
-            overloadApply(f, (Dsymbol s)
+            StringTable funcTypeHash;
+            funcTypeHash._init();
+
+            /* Compute the function signature and insert it in the
+             * hashtable, if not present. This is needed so that
+             * traits(getOverlods, F3, "visit") does not count `int visit(int)`
+             * twice in the following example:
+             *
+             * =============================================
+             * interface F1 { int visit(int);}
+             * interface F2 { int visit(int); void visit(); }
+             * interface F3 : F2, F1 {}
+             *==============================================
+             */
+            void insertInterfaceInheritedFunction(FuncDeclaration fd, Expression e)
+            {
+                bool yes = true;
+                auto funcType = fd.type.toChars();
+                auto len = strlen(funcType);
+                //printf("%s - %s\n", fd.toChars, fd.type.toChars());
+                if (!funcTypeHash.lookup(funcType, len))
```

Comment:
1) Why don't you use D strings and a normal hashmap?
2) `toChars` is a bit of an anti-pattern. Why don't you just insert the `type` directly?

Reply:
Type is a class that does not have `opEquals` and `toHash` defined.